Makefile

```diff
@@ -160,3 +134,83 @@ golangci-lint-fix.%: $(GOLANGCI_LINT)
 
 .PHONY: FORCE # use this to force phony behavior for targets with pattern rules
 FORCE:
+
+##@ Deployment
+
+.PHONY: cluster
+cluster: kind ctlptl ## Create Kind cluster and local registry
+	$(CTLPTL) apply -f ctlptl.yaml
+
+.PHONY: cluster-reset
+cluster-reset: kind ctlptl ## Delete Kind cluster
+	$(CTLPTL) delete -f ctlptl.yaml
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: deploy
+deploy: .gen kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build . | $(KUBECTL) apply -f -
+
+.PHONY: undeploy
+undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build . | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
```

Comment:
I'm confused by this `ignore-not-found` setup. What is the effect if we always use `--ignore-not-found=true`? That seems reasonable to me.

I'm also not sure how I would set `ignore-not-found` in the environment. That isn't a valid shell varname. I don't recall seeing any makefile control vars that don't follow the usual shell var format.

Reply:
E.g. `make undeoloy ignore-not-found=false` - this is something that kubebuilder itself generates. We can skip this, I was hesitant to even add it, let me fix it and simply the target a little bit.

Reply:
Interesting. I haven't seen that kind of make structure before, but it seems potentially useful to keep in mind if we need it at some point (unlikely?).